core/src/OpenMP/Kokkos_OpenMP_Instance.cpp

```diff
@@ -139,13 +141,19 @@ void OpenMPInternal::resize_thread_data(size_t pool_reduce_bytes,
   }
 }
 
+OpenMPInternal *OpenMPInternal::s_singleton = nullptr;
+
 OpenMPInternal &OpenMPInternal::singleton() {
-  static OpenMPInternal *self = nullptr;
-  if (self == nullptr) {
-    self = new OpenMPInternal(get_current_max_threads());
-  }
+  allocate_singleton();
+  return *s_singleton;
+}
+
```

Comment:
I'd just do
```C++
OpenMPInternal &OpenMPInternal::singleton() {
  static OpenMPInternal *self = new OpenMPInternal(get_current_max_threads());
}
```
but really this function doesn't require any changes for what you are trying to achieve here.

Reply:
Outdated.